enterprise/coderd/workspaceproxy.go

```diff
@@ -522,6 +531,14 @@ func (api *API) workspaceProxyRegister(rw http.ResponseWriter, r *http.Request)
 		return
 	}
 
+	if req.Version != buildinfo.Version() {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Version mismatch.",
+			Detail:  fmt.Sprintf("Proxy version %q does not match primary server version %q", req.Version, buildinfo.Version()),
+		})
+		return
+	}
```

Comment:
Strict! 

Can we add something to not check if the version is compiled without a tag? Or if it has a dev tag? Just don't want things to break when I'm testing and might have some conflicts here.

Reply:
It's now disabled in dev builds unless running in go test.